packages/protocol-perf/src/index.ts

```diff
@@ -52,12 +52,19 @@ import type { AbortOptions } from '@libp2p/interfaces'
 const log = logger('libp2p:perf')
 
 export const defaultInit: PerfServiceInit = {
-  protocolName: '/perf/1.0.0',
+  protocolName: '/perf/1.2.0',
   writeBlockSize: BigInt(64 << 10)
 }
 
 export interface PerfService {
-  measurePerformance: (startTime: number, connection: Connection, sendBytes: bigint, recvBytes: bigint, options?: AbortOptions) => Promise<number>
+  measurePerformance: (connection: Connection, sendBytes: bigint, recvBytes: bigint, options?: AbortOptions) => Promise<PerfOutput>
+}
+
+export interface PerfOutput {
+  type: 'intermediary' | 'final'
+  timeSeconds: number
+  uploadBytes: string
+  downloadBytes: string
```

Comment:
Why represent these as `string` and not `number`? Wouldn't `number` suffice in precision?

Reply:
Also this interface exists primarily for output, so it's [only instantiated](https://github.com/libp2p/js-libp2p/pull/2067/files#diff-8bb96d7a36e8cb630880dd75531f3c4517659bf5912d4faaaf2f03f7757a2472R221-R222)f for the purposes of outputting to `stdout` , otherwise the values are being incremented as `bigint`

Reply:
> “Number” would limit us to representing 2^53 bytes (maximum integer in JS, 9007199254740990) which would prevent us from representing values greater than roughly 9KB.

I am confused. `2^53 - 1` which is `9007199254740990` allows us to represent 8 PB (`2^53 ÷ 1024 ÷ 1024 ÷ 1024 ÷ 1024`). What am I missing @SgtPooki?

I am basing the above on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

> Also, a number in json would be converted back to a Number in JS, losing precision in those cases.

Given that we only use JS parsers (NodeJS on both ends) and given that JSON itself does not define a `number` size limit, shouldn't precision be fine until `2^53`? Also note, that high precision is not relevant here.



> Also this interface exists primarily for output, so it's [only instantiated](https://github.com/libp2p/js-libp2p/pull/2067/files#diff-8bb96d7a36e8cb630880dd75531f3c4517659bf5912d4faaaf2f03f7757a2472R221-R222) for the purposes of outputting to `stdout` , otherwise the values are being incremented as `bigint`s

Fine by me. As long as I can parse the values as `number` on the benchmark runner side.

Reply:
The output will is as a result of the `toString()` of a `bigint` and the perf test runner is currently using [MAX_SAFE_NUMBER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) which is a static property of the JS number object, so you will be fine parsing this string as a number @mxinden